packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -955,3 +1001,20 @@ class PipelineLocation {
     return `Stage ${this.stageIndex + 1} Action ${this.action.runOrder} ('${this.stageName}'/'${this.actionName}')`;
   }
 }
+
+/**
+ * Whether two string probably contain the same environment dimension (region or account)
+ *
+ * Used to compare either accounts or regions, and also returns true if both
+ * are unresolved (in which case both are expted to be "current region" or "current account").
+ */
+function sameEnvDimension(a: string, b: string) {
```

Comment:
If you extract `resourceInSameEnvAsItsStack` to a function, like I suggest, I'm pretty sure you'll be able to get rid of this one.